src/foam/nanos/auth/HumanNameTrait.js

```diff
@@ -32,13 +32,17 @@ foam.INTERFACE({
       documentation: 'Full legal name of user. Appends first, middle & last name.',
       transient: true,
       expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
+        return [firstName, middleName, lastName].filter(name => name).join(' ');
       },
       javaGetter: `
-        return ! getMiddleName().equals("")
-          ? getFirstName() + " " + getMiddleName() + " " + getLastName()
-          : getFirstName() + " " + getLastName();
-      `,
+        java.util.List<String> filteredNameList = java.util.Arrays
```

Comment:
This seems a little heavy weight for such a simple thing. On the client we don't care because there's a whole computer for only one user, but we need to share a server with thousands of users.